test/integration/link_dom.cc

```diff
@@ -345,6 +345,31 @@ TEST(DOMLink, Sensors)
   ASSERT_NE(nullptr, thermalCamSensor);
   EXPECT_EQ("my_thermal_camera", thermalCamSensor->Name());
 
+  // Get the segmentation sensor
+  const sdf::Sensor *segmentationSensor =
+    link->SensorByName("segmentation_sensor");
+  ASSERT_NE(nullptr, segmentationSensor);
+  EXPECT_EQ("segmentation_sensor", segmentationSensor->Name());
+  EXPECT_EQ(sdf::SensorType::SEGMENTATION_CAMERA, segmentationSensor->Type());
+  EXPECT_EQ(ignition::math::Pose3d(37, 38, 39, 0, 0, 0),
+            segmentationSensor->RawPose());
+  const sdf::Camera *segmentationCameraSensor =
+    segmentationSensor->CameraSensor();
+  ASSERT_NE(nullptr, segmentationCameraSensor);
+  EXPECT_EQ("my_segmentation_camera", segmentationCameraSensor->Name());
```

Comment:
Can we also test the unique elements for a segmentation sensor (`segmentation_type`)? You'll need to add the unique elements to `test/sdf/sensors.sdf`

Reply:
We discussed that before in the above comments, we agreed that the approach of accessing that types from ign-sensors (via the SDF API) should be fine

here
> I do these configurations in ign-sensors, by getting them through the SDF
> like that: https://github.com/ignitionrobotics/ign-sensors/pull/133/files#diff-3a47a41297adece53d9e25cb71940e30d655d1a491020c1a8fe918a4f1a95ac2R146
> 
> ```c++
> std::string type = sdfElement->Get<std::string>("segmentation_type");
> ```
> 
> is that a bad way to do that ? :D

to add this unique elements, i think I should create a new SDF sensor type for the semgnetation type for example

Reply:
Hmm okay so after looking through the SDF repo for a bit, I actually think that there's a better way than `sdfElement->Get<std::string>` for retrieving the unique element that's associated with a sensor. I think that we should follow the `depth_camera` as an example for creating a camera with a few custom parameters.

If we look at the `depth_camera`, it has a few unique elements on top of all the other `camera` elements: https://github.com/ignitionrobotics/sdformat/blob/sdformat11_11.2.2/sdf/1.8/camera.sdf#L47-L63

These unique elements are accessed via getters/setters in the `sdf::Camera` class:
* https://github.com/ignitionrobotics/sdformat/blob/sdformat11_11.2.2/include/sdf/Camera.hh#L139-L153
* https://github.com/ignitionrobotics/sdformat/blob/sdformat11_11.2.2/include/sdf/Camera.hh#L163-L189

Here's an example of how this looks in sdf (notice `depth_camera` being used in lines 54-59):
* https://github.com/osrf/subt/blob/bff4f4898fa9c2aabeef1b69833d7cf04ee356fa/submitted_models/ctu_cras_norlab_absolem_sensor_config_1/urdf/realsense.gazebo.xacro#L25-L69

---

So, I think we should actually do the following:
1. Add getters/setters to the `sdf::Camera` class for the segmentation and bounding box types
2. Update the camera SDF docs (`sdf/1.9/camera.sdf`) to include unique elements for segmentation and bounding box cameras (see the [depth camera elements](https://github.com/ignitionrobotics/sdformat/blob/sdformat11_11.2.2/sdf/1.8/camera.sdf#L47-L63) as an example). This shouldn't be too difficult - just move the content of your new `sdf/1.9/*.sdf` files over to `sdf/1.9/camera.sdf`. Once everything is moved to `sdf/1.9/camera.sdf`, you can delete the other `sdf/1.9/*.sdf` files that you created and also remove lines [60](https://github.com/ignitionrobotics/sdformat/blob/178ae03abe3f93d1a9eb89b49e68093c52b20b3d/sdf/1.9/sensor.sdf#L60) and [72](https://github.com/ignitionrobotics/sdformat/blob/178ae03abe3f93d1a9eb89b49e68093c52b20b3d/sdf/1.9/sensor.sdf#L72) from `sdf/1.9/sensors.sdf` 
3. Update SDF files that use these sensors (both in this PR and other PRs) to call the unique elements (`segmentation_type` and `box_type`) inside of the `<camera>` element instead of the `<sensor>` element, since we are adding this functionality to the `sdf::Camera` class
4. Update the `sdf::Camera` unit test to cover these new getters/setters (example from the `depth_camera` elements: https://github.com/ignitionrobotics/sdformat/blob/sdformat11_11.2.2/src/Camera_TEST.cc#L45-L55)
5. Double-check all other related PRs that use these sensors and update them to use the newly added `sdf::Camera` APIs as needed

Here's the commit that adds a few unique elements to `depth_camera` that you can use as a reference/example: https://github.com/ignitionrobotics/sdformat/commit/1cbdfe4121f9b4bb51ddfeb0bac9bd6a0153e2a6#

After you make these changes, we should no longer need to use `sdfElement->Get<std::string>` for getting the segmentation and bounding box type elements (it will now be a part of the `sdf::Camera` API). Let me know if you have any questions!